commit 6be35c700f742e911ecedd07fcc43d4439922334
Merge: e37aa63e87bd 520dfe3a3645
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Wed Dec 12 18:07:07 2012 -0800

    Merge git://git.kernel.org/pub/scm/linux/kernel/git/davem/net-next
    
    Pull networking changes from David Miller:
    
    1) Allow to dump, monitor, and change the bridge multicast database
       using netlink.  From Cong Wang.
    
    2) RFC 5961 TCP blind data injection attack mitigation, from Eric
       Dumazet.
    
    3) Networking user namespace support from Eric W. Biederman.
    
    4) tuntap/virtio-net multiqueue support by Jason Wang.
    
    5) Support for checksum offload of encapsulated packets (basically,
       tunneled traffic can still be checksummed by HW).  From Joseph
       Gasparakis.
    
    6) Allow BPF filter access to VLAN tags, from Eric Dumazet and
       Daniel Borkmann.
    
    7) Bridge port parameters over netlink and BPDU blocking support
       from Stephen Hemminger.
    
    8) Improve data access patterns during inet socket demux by rearranging
       socket layout, from Eric Dumazet.
    
    9) TIPC protocol updates and cleanups from Ying Xue, Paul Gortmaker, and
       Jon Maloy.
    
    10) Update TCP socket hash sizing to be more in line with current day
        realities.  The existing heurstics were choosen a decade ago.
        From Eric Dumazet.
    
    11) Fix races, queue bloat, and excessive wakeups in ATM and
        associated drivers, from Krzysztof Mazur and David Woodhouse.
    
    12) Support DOVE (Distributed Overlay Virtual Ethernet) extensions
        in VXLAN driver, from David Stevens.
    
    13) Add "oops_only" mode to netconsole, from Amerigo Wang.
    
    14) Support set and query of VEB/VEPA bridge mode via PF_BRIDGE, also
        allow DCB netlink to work on namespaces other than the initial
        namespace.  From John Fastabend.
    
    15) Support PTP in the Tigon3 driver, from Matt Carlson.
    
    16) tun/vhost zero copy fixes and improvements, plus turn it on
        by default, from Michael S. Tsirkin.
    
    17) Support per-association statistics in SCTP, from Michele
        Baldessari.
    
    And many, many, driver updates, cleanups, and improvements.  Too
    numerous to mention individually.
    
    * git://git.kernel.org/pub/scm/linux/kernel/git/davem/net-next: (1722 commits)
      net/mlx4_en: Add support for destination MAC in steering rules
      net/mlx4_en: Use generic etherdevice.h functions.
      net: ethtool: Add destination MAC address to flow steering API
      bridge: add support of adding and deleting mdb entries
      bridge: notify mdb changes via netlink
      ndisc: Unexport ndisc_{build,send}_skb().
      uapi: add missing netconf.h to export list
      pkt_sched: avoid requeues if possible
      solos-pci: fix double-free of TX skb in DMA mode
      bnx2: Fix accidental reversions.
      bna: Driver Version Updated to 3.1.2.1
      bna: Firmware update
      bna: Add RX State
      bna: Rx Page Based Allocation
      bna: TX Intr Coalescing Fix
      bna: Tx and Rx Optimizations
      bna: Code Cleanup and Enhancements
      ath9k: check pdata variable before dereferencing it
      ath5k: RX timestamp is reported at end of frame
      ath9k_htc: RX timestamp is reported at end of frame
      ...

diff --cc arch/arm/boot/dts/am33xx.dtsi
index 20a3f29a6bfe,a4615b44f25e..c2f14e875eb6
--- a/arch/arm/boot/dts/am33xx.dtsi
+++ b/arch/arm/boot/dts/am33xx.dtsi
@@@ -222,121 -210,53 +222,168 @@@
  			interrupts = <91>;
  		};
  
 +		dcan0: d_can@481cc000 {
 +			compatible = "bosch,d_can";
 +			ti,hwmods = "d_can0";
 +			reg = <0x481cc000 0x2000>;
 +			interrupts = <52>;
 +			status = "disabled";
 +		};
 +
 +		dcan1: d_can@481d0000 {
 +			compatible = "bosch,d_can";
 +			ti,hwmods = "d_can1";
 +			reg = <0x481d0000 0x2000>;
 +			interrupts = <55>;
 +			status = "disabled";
 +		};
 +
 +		timer1: timer@44e31000 {
 +			compatible = "ti,omap2-timer";
 +			reg = <0x44e31000 0x400>;
 +			interrupts = <67>;
 +			ti,hwmods = "timer1";
 +			ti,timer-alwon;
 +		};
 +
 +		timer2: timer@48040000 {
 +			compatible = "ti,omap2-timer";
 +			reg = <0x48040000 0x400>;
 +			interrupts = <68>;
 +			ti,hwmods = "timer2";
 +		};
 +
 +		timer3: timer@48042000 {
 +			compatible = "ti,omap2-timer";
 +			reg = <0x48042000 0x400>;
 +			interrupts = <69>;
 +			ti,hwmods = "timer3";
 +		};
 +
 +		timer4: timer@48044000 {
 +			compatible = "ti,omap2-timer";
 +			reg = <0x48044000 0x400>;
 +			interrupts = <92>;
 +			ti,hwmods = "timer4";
 +			ti,timer-pwm;
 +		};
 +
 +		timer5: timer@48046000 {
 +			compatible = "ti,omap2-timer";
 +			reg = <0x48046000 0x400>;
 +			interrupts = <93>;
 +			ti,hwmods = "timer5";
 +			ti,timer-pwm;
 +		};
 +
 +		timer6: timer@48048000 {
 +			compatible = "ti,omap2-timer";
 +			reg = <0x48048000 0x400>;
 +			interrupts = <94>;
 +			ti,hwmods = "timer6";
 +			ti,timer-pwm;
 +		};
 +
 +		timer7: timer@4804a000 {
 +			compatible = "ti,omap2-timer";
 +			reg = <0x4804a000 0x400>;
 +			interrupts = <95>;
 +			ti,hwmods = "timer7";
 +			ti,timer-pwm;
 +		};
 +
 +		rtc@44e3e000 {
 +			compatible = "ti,da830-rtc";
 +			reg = <0x44e3e000 0x1000>;
 +			interrupts = <75
 +				      76>;
 +			ti,hwmods = "rtc";
 +		};
 +
 +		spi0: spi@48030000 {
 +			compatible = "ti,omap4-mcspi";
 +			#address-cells = <1>;
 +			#size-cells = <0>;
 +			reg = <0x48030000 0x400>;
 +			interrupt = <65>;
 +			ti,spi-num-cs = <2>;
 +			ti,hwmods = "spi0";
 +			status = "disabled";
 +		};
 +
 +		spi1: spi@481a0000 {
 +			compatible = "ti,omap4-mcspi";
 +			#address-cells = <1>;
 +			#size-cells = <0>;
 +			reg = <0x481a0000 0x400>;
 +			interrupt = <125>;
 +			ti,spi-num-cs = <2>;
 +			ti,hwmods = "spi1";
 +			status = "disabled";
 +		};
 +
 +		usb@47400000 {
 +			compatible = "ti,musb-am33xx";
 +			reg = <0x47400000 0x1000	/* usbss */
 +			       0x47401000 0x800		/* musb instance 0 */
 +			       0x47401800 0x800>;	/* musb instance 1 */
 +			interrupts = <17		/* usbss */
 +				      18		/* musb instance 0 */
 +				      19>;		/* musb instance 1 */
 +			multipoint = <1>;
 +			num-eps = <16>;
 +			ram-bits = <12>;
 +			port0-mode = <3>;
 +			port1-mode = <3>;
 +			power = <250>;
 +			ti,hwmods = "usb_otg_hs";
 +		};
++
+ 		mac: ethernet@4a100000 {
+ 			compatible = "ti,cpsw";
+ 			ti,hwmods = "cpgmac0";
+ 			cpdma_channels = <8>;
+ 			ale_entries = <1024>;
+ 			bd_ram_size = <0x2000>;
+ 			no_bd_ram = <0>;
+ 			rx_descs = <64>;
+ 			mac_control = <0x20>;
+ 			slaves = <2>;
+ 			cpts_active_slave = <0>;
+ 			cpts_clock_mult = <0x80000000>;
+ 			cpts_clock_shift = <29>;
+ 			reg = <0x4a100000 0x800
+ 			       0x4a101200 0x100>;
+ 			#address-cells = <1>;
+ 			#size-cells = <1>;
+ 			interrupt-parent = <&intc>;
+ 			/*
+ 			 * c0_rx_thresh_pend
+ 			 * c0_rx_pend
+ 			 * c0_tx_pend
+ 			 * c0_misc_pend
+ 			 */
+ 			interrupts = <40 41 42 43>;
+ 			ranges;
+ 
+ 			davinci_mdio: mdio@4a101000 {
+ 				compatible = "ti,davinci_mdio";
+ 				#address-cells = <1>;
+ 				#size-cells = <0>;
+ 				ti,hwmods = "davinci_mdio";
+ 				bus_freq = <1000000>;
+ 				reg = <0x4a101000 0x100>;
+ 			};
+ 
+ 			cpsw_emac0: slave@4a100200 {
+ 				/* Filled in by U-Boot */
+ 				mac-address = [ 00 00 00 00 00 00 ];
+ 			};
+ 
+ 			cpsw_emac1: slave@4a100300 {
+ 				/* Filled in by U-Boot */
+ 				mac-address = [ 00 00 00 00 00 00 ];
+ 			};
 -
+ 		};
  	};
  };